calculator.js

```diff
@@ -5,164 +5,173 @@ let lastButtonClicked = null;
-    const currentValueOfResult = result.textContent;
-    const newValue = currentValueOfResult + 8;
-    console.log(parseInt(newValue));
-    result.textContent = newValue;
-    lastButtonClicked = "Number";
-});
-
-button9.addEventListener("click", function() {
-    if (lastButtonClicked === "Operator") result.textContent = "";
-    const currentValueOfResult = result.textContent;
-    const newValue = currentValueOfResult + 9;
-    console.log(parseInt(newValue));
-    result.textContent = newValue;
-    lastButtonClicked = "Number";
-});
-
-periodButton.addEventListener("click", function() {
-    const currentValueOfResult = result.textContent;
-    if (currentValueOfResult.includes('.') !== true)
-        result.textContent = currentValueOfResult + ".";
-});
-
-negativeValueButton.addEventListener("click", function() {
-    const currentValueOfResult = result.textContent;
-    if (currentValueOfResult !== "" && currentValueOfResult[0] !== "-") { result.textContent = "-" + currentValueOfResult };
-    if (currentValueOfResult !== "" && currentValueOfResult[0] === "-") { result.textContent = currentValueOfResult.slice(1) };
-});
-
-function evaluate() {
-    if (lastValue && result.textContent && operator !== null) {
-        let firstOperand = parseFloat(lastValue);
-        let secondOperand = parseFloat(result.textContent);
-        let _result;
-        if (operator === "+") _result = firstOperand + secondOperand;
-        if (operator === "-") _result = firstOperand - secondOperand;
-        if (operator === "*") _result = firstOperand * secondOperand;
-        if (operator === "/") _result = firstOperand / secondOperand;
-        result.textContent = _result;
-        if (operator === "/" && secondOperand === 0)
-            result.textContent = 'Cannot divide by zero';
-    }
-};
-
-addingButton.addEventListener("click", function() {
-    evaluate();
-    operator = "+";
-    lastButtonClicked = "Operator";
-    lastValue = result.textContent;
-});
-
-subtractingButton.addEventListener("click", function() {
-    evaluate();
-    operator = "-";
-    lastButtonClicked = "Operator";
-    lastValue = result.textContent;
-});
-
-multiplyingButton.addEventListener("click", function() {
-    evaluate();
-    operator = "*";
-    lastButtonClicked = "Operator";
-    lastValue = result.textContent;
-});
-
-dividingButton.addEventListener("click", function() {
-    evaluate();
-    operator = "/";
-    lastButtonClicked = "Operator";
-    lastValue = result.textContent;
-});
-
-equalButton.addEventListener("click", function() {
-    evaluate();
-});
\ No newline at end of file
+
+numBtnElem.forEach(function(button) {
+            button.addEventListener('click', function() {
+                    (lastButtonClicked === "Operator") result.textContent = "";
+                    const currentValueOfResult = result.textContent;
+                    const newValue = currentValueOfResult + Number;
```

Comment:
Instead of using `Number` which I don't where is coming from 
you should get the value of the button from the dataset of the button

Reply:
In the same way I used Number to assign a value for lastButtonClicked variable I thought let me use it here also!!